supplementary/alica_ros1/alica_ros_turtlesim/test/test_main.cpp

```diff
@@ -0,0 +1,183 @@
+                break;                                                                                                                                         \
+            }                                                                                                                                                  \
+            past_pose = turtle.getPose();                                                                                                                      \
+        } while (duration_cast<milliseconds>(system_clock::now() - start).count() < ms);                                                                       \
+        EXPECT_TRUE(turtle.isPoseEqual(past_pose, 0.01));                                                                                                      \
+    } while (0)
+
+// class to handle callbacks
+class TurtlePosition
+{
+public:
+    TurtlePosition() = default;
+
+    TurtlePosition(const double x, const double y, const double theta = 0)
+    {
+        _pose.x = x;
+        _pose.y = y;
+        _pose.theta = theta;
+    }
+
+    void poseCallback(const turtlesim::Pose::ConstPtr msg)
+    {
+        _pose.x = msg->x;
+        _pose.y = msg->y;
+        _pose.theta = msg->theta;
+    }
+
+    turtlesim::Pose getPose() const { return _pose; }
+
+    bool isNear(const turtlesim::Pose& desired_pose, const turtlesim::Pose& delta) const
+    {
+        assert(delta.x > 0);
+        assert(delta.y > 0);
+
+        return std::fabs(_pose.x - desired_pose.x) < delta.x && std::fabs(_pose.y - desired_pose.y) < delta.y;
+    }
+
+    double distance(const TurtlePosition& turtle) { return sqrt(pow(turtle._pose.x - _pose.x, 2) + pow(turtle._pose.y - _pose.y, 2)); }
+
+    bool isPoseEqual(const turtlesim::Pose& other_pose, double tolerance)
+    {
+        return std::fabs(other_pose.x - _pose.x) + std::fabs(other_pose.y - _pose.y) + std::fabs(other_pose.theta - _pose.theta) < tolerance;
+    }
+
+private:
+    turtlesim::Pose _pose;
+};
+
+TEST(AlicaTurtlesimTest, destinationTest)
+{
+    ros::NodeHandle nh;
+    ros::Publisher join_formation_turtle_1_pub = nh.advertise<std_msgs::Empty>("/turtle1/join_formation", 10);
+    ros::Publisher join_formation_turtle_2_pub = nh.advertise<std_msgs::Empty>("/turtle2/join_formation", 10);
+    ros::Publisher join_formation_turtle_3_pub = nh.advertise<std_msgs::Empty>("/turtle3/join_formation", 10);
+    ros::Publisher join_formation_turtle_4_pub = nh.advertise<std_msgs::Empty>("/turtle4/join_formation", 10);
+
+    // wait for subscribers to latch
+    RUN_UNTIL_EXPECT_EQ(join_formation_turtle_1_pub.getNumSubscribers(), 1U, 10000);
+    RUN_UNTIL_EXPECT_EQ(join_formation_turtle_2_pub.getNumSubscribers(), 1U, 10000);
+    RUN_UNTIL_EXPECT_EQ(join_formation_turtle_3_pub.getNumSubscribers(), 1U, 10000);
+    RUN_UNTIL_EXPECT_EQ(join_formation_turtle_4_pub.getNumSubscribers(), 1U, 10000);
```

Comment:
Probably these can be ASSERTs just to shortcut if there is problem with test setup

Please disable from running this test in CI